src/Classes/Particle/Linker.ts

```diff
@@ -1,29 +1,23 @@
 import type { Container } from "../Container";
 import { ColorUtils } from "../Utils/ColorUtils";
-import { Utils } from "../Utils/Utils";
-import type { ICoordinates } from "../../Interfaces/ICoordinates";
 import { Constants } from "../Utils/Constants";
-import type { IParticle } from "../../Interfaces/IParticle";
+import { Particle } from "../Particle";
 
 export class Linker {
-	public static link(p1: IParticle, p2: IParticle, container: Container): void {
-		const pos1: ICoordinates = {
-			x: p1.position.x + p1.offset.x,
-			y: p1.position.y + p1.offset.y,
-		};
-		const pos2: ICoordinates = {
-			x: p2.position.x + p2.offset.x,
-			y: p2.position.y + p2.offset.y,
-		};
-		const dist = Utils.getDistanceBetweenCoordinates(pos1, pos2);
-		const optOpacity = p1.particlesOptions.lineLinked.opacity;
-		const optDistance = p1.lineLinkedDistance ?? container.retina.lineLinkedDistance;
-
-		/* draw a line between p1 and p2 if the distance between them is under the config distance */
-		if (dist <= optDistance) {
-			const opacityLine = optOpacity - (dist * optOpacity) / optDistance;
+	public static link(p1: Particle, container: Container): void {
+		const options = container.options;
+		const optOpacity = options.particles.lineLinked.opacity;
+		const optDistance = container.retina.lineLinkedDistance;
```

Comment:
this value should be p1.lineLinkedDistance ?? container.retina.lineLinkedDistance